alerts.php

```diff
@@ -8,9 +8,7 @@
 
 $templatelist = 'myalerts_alert_row_popup,myalerts_alert_row_popup_no_alerts,myalerts_modal_content';
 
-if ($_GET['action'] == 'modal') {
-	defined('NO_ONLINE') or define('NO_ONLINE', 1);
-}
+defined('NO_ONLINE') or define('NO_ONLINE', 1);
```

Comment:
This change will prevent any page in `alerts.php` from being tracked on the Who's Online list.

The removed code has been working correctly on my forum for over a year.

Reply:
> This change will prevent any page in `alerts.php` from being tracked on the Who's Online list.

I didn't notice that MyAlerts actually build a WOL location, that should answer my 3rd question.

In that case, we should refactor `myalerts_online_location` hook:
https://github.com/MyBBStuff/MyAlerts/blob/19714c73b44e161081d9877677a2b15dae16be64/inc/plugins/myalerts.php#L858-L876

Since MyAlerts is not using `usercp` anymore (can anyone confirm everything already moved to alerts.php?), we can get rid of the check.

```php
$inAlertsPage = $plugin_array['user_activity']['activity'] == 'alerts'; 
```
`$inAlertsPage` is always set to `false`. I can't figure it out how `activity` set to `alerts` but I can see how it set to `unknown`: [functions_online.php#L565-L567](https://github.com/mybb/mybb/blob/35a9edf9b7652940e723098197d853bc173a7880/inc/functions_online.php#L565-L567). The location name is never set by MyAlerts then?

> The removed code has been working correctly on my forum for over a year.

Isn't it impossible for that code to execute since the expression is always `false`? (If used normally, because of 17643861578e29755029425216bcdd0af74e72d2). Any chance you disable the PHP notice?

Reply:
If you want to fix the PHP warning, simply use:

```php
if (isset($_GET['action']) && $_GET['action'] == 'modal') {
...
```

Reply:
Sure, that should fix the PHP warning, but it's a dead code. Since 17643861578e29755029425216bcdd0af74e72d2, `$_GET['action']` should never set to `modal` anymore. The right way should be:

```php
if (isset($_GET['modal']) && $_GET['modal'] == '1') {
...
```

But then it still doesn't solve the "Unknown Location" because the real problem is in `myalerts_online_location` function. It hasn't refactored for the change, and that's what my latest commit addressed for.

Reply:
Okay thanks for catching me up. I still use an older version of MyAlerts because, well, it works.

Still, the changes above are the appropriate solution for the problem (using the code you just posted instead of removing it entirely from `alerts.php`

Reply:
I'm afraid it doesn't solve anything, I'm not even sure why the code exists.

The real problem lies here, this code below will never ever set `$plugin_array['location_name']` because both `$inUserCpAlerts` and `$inAlertsPage` are always equal to `false`, again because it hasn't refactored for the change.
https://github.com/MyBBStuff/MyAlerts/blob/19714c73b44e161081d9877677a2b15dae16be64/inc/plugins/myalerts.php#L858-L876

Reply:
Just to be clear, I am saying that despite the issues, the problem is not solved by removing that conditional from `alerts.php`